.circleci/config.yml

```diff
@@ -34,8 +23,37 @@ jobs:
           paths:
             - node_modules
             - app/node_modules
-          key: v1-dependencies-{{ checksum "yarn.lock" }}-{{ checksum "app/yarn.lock"}}
+            - ../.cache/Cypress
+          key: dependencies-{{ checksum "yarn.lock" }}-{{ checksum "app/yarn.lock"}}
 
       # run tests!
       - run: yarn eslint
       - run: yarn test-chrome-headless
+      - run:
+          name: Build the app so that server can be started
+          command: cd app && yarn build && cd ../
+      - run:
+          name: Start server with correct env for cypress tests
+          command: >
+            yarn generate-cert && ADALITE_ENABLE_HTTPS=true
+            ADALITE_SERVER_URL='https://localhost:3000' PORT=3000
+            ADALITE_BLOCKCHAIN_EXPLORER_URL='https://explorer-testnet.adalite.io'
+            ADALITE_NETWORK='MARY_TESTNET' ADALITE_ENABLE_SERVER_MOCKING_MODE=true
+            ADALITE_MOCK_TX_SUBMISSION_SUCCESS=true ADALITE_MOCK_TX_SUMMARY_SUCCESS=true
+            yarn start-server
+          wait-on: https://localhost:3000
+          background: true
+      - run:
+          name: Run Cypress tests
+          store_artifacts: true
+          command: yarn cypress:run
+      - store_artifacts:
+          path: app/cypress/videos
+      - store_artifacts:
+          path: app/cypress/screenshots
+
+workflows:
+  version: 2
+  workflow:
+    jobs:
+      - build
```

Comment:
A bit unrelated to the cypress tests, but I thought about splitting this workflow into multiple jobs, 
Problem: 

if yarn audit fails, all the other steps are automatically canceled, in that case ( and I guess it happened to all of us at some point) I go ahead and run yarn audit locally, fix the problem just to find out after another push that now eslint is failing, so I go ahead and fix that, just to find out after another push that but some test are failing. :smile: 
Of course I can always run this locally before every commit tho I would prefer to rely on cicd to tell me, and in that case I would like it to tell me everything that's wrong.

Solution:
I don't really see into ci-ci that much but I guess splitting the workflow into multiple jobs, (build, static checks (like eslint), unit tests, integration tests) would solve the issue. (maybe not I am really not sure)

I guess we can look into in a another PR but since you are already working with and it might be pretty simple.

Reply:
I'll create a separate issue.